mlflow/xgboost/__init__.py

```diff
@@ -361,8 +362,31 @@ def predict(
         Returns:
             Model predictions.
         """
+        import xgboost as xgb
+
         predict_fn = _wrapped_xgboost_model_predict_fn(self.xgb_model)
-        return predict_fn(dataframe, **(params or {}))
+        params = params or {}
+        # filter is applied inside predict_fn wrapper for xgb.Booster
+        if not isinstance(self.xgb_model, xgb.Booster):
+            params = _filter_allowed_kwargs(predict_fn, params)
```

Comment:
Does this exclude recognized parameters if the `predict_fn` signature looks like `predict_fn(*args, **kwargs)`?

Reply:
Yes, updated to avoid exclusion in such case